regen-network/regen-web
# feat: project forms validation

## Description

Closes: regen-network/regen-registry#1501

- uses Yup for project forms validation (we'll migrate to zod as part of https://github.com/regen-network/regen-registry/issues/1451), except on the `ProjectMetadataForm` since the validation depends on the credit class so in this case, it uses SHACL
- also fixes an issue with `FieldFormControl` https://github.com/jaredpalmer/formik/issues/2083



---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...


#### Setting up backport PRs

After merging your PR to `dev`, set up backports by doing the following:

1. If your branch does not have merge commits, add the following comment to
   your PR:

- If PR is a new feature: `@Mergifyio backport redwood hambach`.
- If PR is a bug fix: `@Mergifyio backport redwood hambach master`.

2. If your branch does have merge commits:

   a. Pull latest `dev`, `hambach` and `redwood` branches

   b. Create new branches for backports and merge `dev` (replace `<PR#>` with your PR #)

   ```
   git checkout -b hambach-backport-<PR#> hambach
   git merge dev
   git push origin hambach-backport-<PR#>
   git checkout -b redwood-backport-<PR#> redwood
   git merge dev
   git push origin redwood-backport-<PR#>`
   ```

   c. Open new PRs in regen-web targeting `hambach` and `redwood`, respectively.

   d. If PR is a bug fix repeat the same process for `master` branch.

### How to test

Create a new project from https://deploy-preview-1794--regen-registry.netlify.app/ecocredits/projects (using our test addr regen1df675r9vnf7pdedn4sf26svdsem3ugavgxmy46) and go through the different forms to make sure the validation is working correctly.

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...
